lib/configparser.ex

```diff
@@ -266,8 +266,8 @@ defmodule ConfigParser do
   end
 
   # Returns true if the line appears to be a comment
-  @hash_comment_regex ~r{#.*}
-  @semicolon_comment_regex ~r{;.*}
+  @hash_comment_regex ~r{^#.*}
+  @semicolon_comment_regex ~r{^;.*}
```

Comment:
This is a pretty good idea. I was trying to keep the changes in this PR to a minimum but I'll give this a shot.